### ISSUE TITLE
Uploads broken

### ISSUE THREAD

**Issue by srittau** [typeshed-internal/stub_uploader#68]
Currently, it seems that PyPI uploads are broken. See for example these two runs:

* https://github.com/typeshed-internal/stub_uploader/actions/runs/3228142124/jobs/5283851894
* https://github.com/typeshed-internal/stub_uploader/actions/runs/3247524346/jobs/5327604962

But none of the stubs (regex, colorama, paho-mqtt, redis) have recent uploads.

**Comment by srittau** [comment]
Manual updates are also broken: https://github.com/typeshed-internal/stub_uploader/actions/runs/3249872314/jobs/5332782779

**Comment by JelleZijlstra** [comment]
I suspect there's something wrong with the `sort_by_dependency()` function; that would explain why nothing happens after it prints what packages it would upload. I can't reproduce the problem locally though.

**Comment by JelleZijlstra** [comment]
I just triggered a manual run and it worked: https://pypi.org/project/types-regex/#history got uploaded. Not sure what changed.

**Comment by hauntsaninja** [comment]
Sorry for the trouble! I just merged something to typeshed, so hopefully we'll have another datapoint soon. One more note is I'll have limited availability this weekend. Feel free to revert #61 if need be (looks to me like things were working before that, at least).

@JelleZijlstra I might be confused, but didn't we print the "Building and uploading stubs for" message after `sort_by_dependency`?

**Comment by JelleZijlstra** [comment]
> @JelleZijlstra I might be confused, but didn't we print the `Building and uploading stubs for: redis, colorama, paho-mqtt` message after `sort_by_dependency`?

We do, but the printed output is the input to `sort_by_dependency`: https://github.com/typeshed-internal/stub_uploader/blob/f3e24f676bac5ca892cf209f40eb9fa78e270cf2/stub_uploader/upload.py#L25

**Comment by hauntsaninja** [comment]
So https://github.com/typeshed-internal/stub_uploader/actions/runs/3252136124/jobs/5338000226 seems to have worked

**Comment by srittau** [comment]
I also (successfully) triggered the rebuild of redis, colorama, and paho-mqtt. I would still like to keep this open for a bit. I'm wary of a bug that fixes itself.

**Comment by hauntsaninja** [comment]
Looks like things have continued to work, as far as I can tell.

> We do, but the printed output is the input to sort_by_dependency

I think that's true on main, but false on the commit we saw the issue (got changed in #69), see: https://github.com/typeshed-internal/stub_uploader/blob/0c5e1d04f82fbd2e4d94925aba42dc5f6586cf2e/stub_uploader/upload_changed.py#L35

I'm really a little mystified. Assuming there was an issue that is now fixed, #69 seems most likely to have fixed it.
